gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -151,7 +173,16 @@ void onSignedOut() {
 
   private void onSignedIn() {
     hideProgressDialog();
-    navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+    if (termsState.getValue() == LoadState.NOT_FOUND) {
+      localValueStore.setTermsAccepted(true);
+      navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+    } else {
+      if (signInProgressDialogVisibility.getValue() == null) {
```

Comment:
@HemanParbhakar Why is this necessary? Are you using the visibility of the sign in dialog as a proxy for something else?